frb/galaxies/galfit.py

```diff
@@ -0,0 +1,463 @@
+"""
+Automate the GALFIT analysis procedure. GALFIT
+is a free, closed-source software that can be found
+here:
+https://users.obs.carnegiescience.edu/peng/work/galfit/galfit.html
+
+Here is the user manual:
+https://users.obs.carnegiescience.edu/peng/work/galfit/README.pdf
+
+This module is mainly for a python wrapper to automate
+using GALFIT. It also contains additional functions
+to parse the output automatically.
+"""
+
+import numpy as np
+import pandas as pd
+import sys, os, subprocess
+import warnings
+from astropy.io import fits
+from astropy import units as u
+from astropy.stats import sigma_clipped_stats
+from astropy.table import Table
+from astropy.wcs import WCS
+import re
+
+def write_cutout(cutout, filename = "cutout.fits", overwrite=False):
+    """
+    Takes an astropy Cutout2D
+    object and writes it to disk.
+    Args:
+        cutout (Cutout2D): Prefaerably
+            has WCS info in it.
+        filename (str, optional): FITS
+            file into which data is saved.
+            Defaults to "cutout.fits" in
+            the current working directory.
+        overwrite (bool, optional): Do you
+            want to overwrite filename if
+            it already exists?
+    Returns:
+    """
+    hdr = cutout.wcs.to_header()
+    imghdu = fits.PrimaryHDU(cutout.data, hdr)
+    hdulist = fits.HDUList([imghdu])
+    hdulist.writeto(filename, overwrite=overwrite)
+    return
+
+def _genconf(imgfile:str, psffile:str,
+            configfile:str=None, outdir:str=None, outfile:str=None,
```

Comment:
whoa!  I've never seen this kind of arg list with defined variable types.

Reply:
Yeah, it's a neat documentation tool someone showed on the astrograds' Slack channel.